File: backend/app/schemas/threat.py
from datetime import datetime
from typing import Optional, List, Dict
from uuid import UUID
from pydantic import BaseModel, ConfigDict
from app.models.threat import ThreatSeverity, ThreatType, ThreatStatus


class ThreatBase(BaseModel):
    scan_result_id: UUID
    threat_type: ThreatType
    severity: ThreatSeverity
    rule_name: str
    description: str
    status: ThreatStatus = ThreatStatus.NEW


class ThreatRecordCreate(ThreatBase):
    pass


class ThreatRecordUpdateStatus(BaseModel):
    status: ThreatStatus


class ThreatRecordOut(ThreatBase):
    id: UUID
    detected_at: datetime
    file_name: Optional[str] = None
    full_path: Optional[str] = None
    sha256: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ThreatSeverityCount(BaseModel):
    CRITICAL: int = 0
    HIGH: int = 0
    MEDIUM: int = 0
    LOW: int = 0
    INFO: int = 0


class ThreatStatsOut(BaseModel):
    total_threats: int
    open_threats: int
    resolved_threats: int
    false_positives: int
    quarantined: int
    severity_breakdown: ThreatSeverityCount
    threat_type_breakdown: Dict[str, int]

File: frontend/src/types/threat.ts
export type ThreatSeverity = "CRITICAL" | "HIGH" | "MEDIUM" | "LOW" | "INFO";

export type ThreatType =
  | "KNOWN_MALWARE"
  | "DOUBLE_EXTENSION"
  | "HIDDEN_EXECUTABLE"
  | "AUTORUN_SCRIPT"
  | "SUSPICIOUS_EXTENSION"
  | "ANOMALOUS_FILE";

export type ThreatStatus = "NEW" | "ACKNOWLEDGED" | "RESOLVED" | "OPEN" | "QUARANTINED" | "FALSE_POSITIVE";

export interface ThreatRecord {
  id: string;
  scan_result_id: string;
  file_name?: string;
  full_path?: string;
  sha256?: string;
  rule_name: string;
  threat_name?: string;
  threat_type: ThreatType | string;
  severity: ThreatSeverity;
  description: string;
  status: ThreatStatus;
  detected_at: string;
  remediation?: string;
}

export interface ThreatSeverityCount {
  CRITICAL: number;
  HIGH: number;
  MEDIUM: number;
  LOW: number;
  INFO: number;
}

export interface ThreatStats {
  total_threats: number;
  open_threats: number;
  resolved_threats: number;
  false_positives: number;
  quarantined: number;
  severity_breakdown: ThreatSeverityCount;
  threat_type_breakdown: Record<string, number>;
}
